interpreter/scope.go

```diff
@@ -0,0 +1,106 @@
+package interpreter
+
+import (
+	"github.com/influxdata/flux/values"
+)
+
+type Scope interface {
+	// Lookup a name in the current scope
+	Lookup(name string) (values.Value, bool)
+
+	// Bind a variable in the current scope
+	Set(name string, v values.Value)
+
+	// Create a new scope by nesting the current scope
```

Comment:
Can you specify how `values.Object` is used? I.e. it  populates the initial values of the nested scope, and may be nil.